sparkprs/jira_api.py

```diff
@@ -12,6 +12,37 @@ def get_jira_client():
                                         app.config['JIRA_PASSWORD']))
 
 
+def start_issue_progress(issue):
+    """
+    Given an issue key, e.g. SPARK-6481, mark the issue "In Progress".
+
+    This will only happen if the issue's initial state is "Open".
+    """
+    jira_client = get_jira_client()
+    issue_info = jira_client.issue(issue)
+    status = issue_info.fields.status.name
+    assignee = issue_info.fields.assignee.name if issue_info.fields.assignee else None
+
+    if status != "Open":
+        logging.warn("Could not start progress on JIRA issue {j}. "
+                     "It's currently in an '{s}' state. Issues must be in an 'Open' state.".format(
+                         j=issue, s=status))
+        return
+
+    # The Apache Spark user needs the issue assigned to itself in order to change
+    # the issue's state.
+    jira_client.assign_issue(issue=issue, assignee='apachespark')
+    transition_id = [transition['id']
+                     for transition in jira_client.transitions(issue)
+                     if transition['name'] == 'Start Progress'][0]
+    # Passing transition by name doesn't work, though it should according to the docs...
+    jira_client.transition_issue(issue=issue, transition=transition_id)
```

Comment:
Just confirmed that the second argument is named `transitionId` in the JIRA client version we use (see lib/jira/client.py)